labs/lab-2-javascript-game/script.js

```diff
@@ -0,0 +1,50 @@
+function startGame() {
+  const play = window.prompt("Do you want to play?");
+  if (play === "yes") {
+    var name = window.prompt("Enter your name:")
+    startCombat(name);
+  }
+}
+
+function getDamage() {
+  const min = 1;
+  const max = 5;
+  return (Math.floor(Math.random() * (max - min + 1) + min));
+}
+
+function startCombat(name) {
+  var winCount = 0;
+  var userHP = 40;
+  var grantHP = 10;
+  var choice;
+
+  while ((winCount < 3) && (userHP > 0)) {
+    choice = window.prompt("Would you like to attack or quit?");
+    if (choice === "attack") {
+      userHP = userHP - getDamage();
+      grantHP = grantHP - getDamage();
+      console.log(name + " has " + userHP + " health left.");
+      console.log("Grant the Mighty Chicken has " + grantHP + " health left.");
+    } else if (choice === "quit") {
+      winCount = 4;
```

Comment:
This confused me a lot reading it the first time -- the user quits and they automatically win 4 times?

I had to read a looong way into the rest of the function before I found out why.

It's less than ideal that `winCount` is tracking the number of wins AND ALSO whether the user has quit prematurely. Perhaps use two variables?